fv3core/testing/validation.py

```diff
@@ -27,11 +26,38 @@ class SelectivelyValidated:
         """
 
         def __init__(self, *args, **kwargs):
+
+            selective_arg_names = []
+            origin_domain_funcs = []
```

Comment:
Instead of these lists, can you use a Mapping[str, Callable] in the argument above and do `for arg_name, func in name_to_function.items():`?